src/metric/dcg_calculator.cpp

```diff
@@ -70,6 +71,10 @@ void DCGCalculator::Init(const std::vector<double>& input_label_gain) {
         std::sscanf(line.c_str(),"%d,%d,%lf", &i, &j, &bias);
         position_bias_lookup_[std::make_pair(i, j)] = bias;
         position_bias_lookup_[std::make_pair(j, i)] = 1.0 / bias;
+        //find the max index in the position bias lookup table
+        if (((i > position_bias_lookup_max_) || (j > position_bias_lookup_max_)) && (bias > 0.00001)){
```

Comment:
the bias > 0.00001 was included because I think at the beginning/end of the csv file we get i and j that are unpredictable / not actually rows in the csv but their bias is 0. might be a cleaner way to check for this.